R/bind.r

```diff
@@ -124,10 +126,18 @@ bind_rows <- function(..., .id = NULL) {
 
 #' @export
 #' @rdname bind
-bind_cols <- function(...) {
+bind_cols <- function(..., .name_repair = "unique") {
   x <- flatten_bindable(dots_values(...))
   out <- cbind_all(x)
-  tibble::repair_names(out)
+
+  # Using `.name_repair` but then gymnastics to keep
+  # attributes in case cbind_all did not return a tibble
+  repaired <- as_tibble(out, .name_repair = .name_repair)
```

Comment:
Perhaps this, and remove gymnastics? Maybe it's worth to implement an internal name repair function that constructs a 0-row tibble to invoke the repair on?

```suggestion
  names(repaired) <- names(as_tibble(out, .name_repair = .name_repair))
```

Reply:
Thanks. Done now.